.github/workflows/pre-merge-checks.yml

```diff
@@ -30,8 +30,42 @@ jobs:
       - name: Run Spotless
         run: ./gradlew --no-daemon spotlessCheck
 
-  run-tests:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - name: Build Cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            .gradle/
+            */build/
+            !*/build/reports
+            !*/build/jacoco
+          key: ${{ runner.os }}-build-${{ github.sha }}
+      - name: Run Build
+        run: ./gradlew --parallel build -x test -x account-management-integration-tests:test -x spotlessApply -x spotlessCheck
```

Comment:
Why are we removing spotless?

Reply:
Have we looked at setting up Gradle parallelisation in the `gradle.properties`, as then it works locally, too?:

```
org.gradle.parallel=true
```

Also maybe JUnit5 parallelisation too, if not already done?

(via https://www.jvt.me/posts/2021/03/11/gradle-speed-parallel/)

Reply:
We're not, the spotless check happens in the "style-checks" job, we just exclude from the build and test jobs to save ourselves some seconds.

Reply:
Sadly, on the `org.gradle.parallel=true` thing, I did consider it, but this could cause us some issues with the integration tests (currently). When running locally, using the `pre-commit.sh` it uses the `--parallel` switch for the build and unit tests, but not for integration tests.

However, I believe its all a little moot as I believe we only get one CPU in GitHub actions, so it can't even run the unit tests in parallel, I actually meant to take that switch out.

Reply:
Hmm that's a shame. Hopefully we can solve that Integration Test issue so we can re-enable parallelisation everywhere - I've suffered in the past of trying to unpick it after many months of the codebase implicitly assuming that tests execute sequentially

Reply:
I was wrong, its actually 2-cpu workers, I leave the flag there for the mo.